pyartifactory/objects/user.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+import ast
+import logging
+from typing import List
+
+import requests
+
+from pyartifactory.exception import ArtifactoryError, UserAlreadyExistsError, UserNotFoundError
+from pyartifactory.models.user import NewUser, SimpleUser, User, UserResponse
+from pyartifactory.objects.object import ArtifactoryObject
+
+logger = logging.getLogger("pyartifactory")
+
+
+class ArtifactoryUser(ArtifactoryObject):
+    """Models an artifactory user."""
+
+    _uri = "security/users"
+
+    def create(self, user: NewUser) -> UserResponse:
+        """
+        Create user
+        :param user: NewUser object
+        :return: User
+        """
+        username = user.name
+        try:
+            self.get(username)
+            logger.error("User %s already exists", username)
+            raise UserAlreadyExistsError(f"User {username} already exists")
+        except UserNotFoundError:
+            data = user.model_dump()
+            data["password"] = user.password.get_secret_value()
+            self._put(f"api/{self._uri}/{username}", json=data)
+            logger.debug("User %s successfully created", username)
+            return self.get(user.name)
+
+    def get(self, name: str) -> UserResponse:
+        """
+        Read user from artifactory. Fill object if exist
+        :param name: Name of the user to retrieve
+        :return: UserModel
+        """
+        try:
+            response = self._get(f"api/{self._uri}/{name}")
+            logger.debug("User %s found", name)
+            return UserResponse(**response.json())
+        except requests.exceptions.HTTPError as error:
+            if error.response.status_code in (404, 400):
+                logger.error("User %s does not exist", name)
+                raise UserNotFoundError(f"{name} does not exist")
+            raise ArtifactoryError from error
+
+    def list_all(self) -> List[SimpleUser]:
+        """
+        Lists all the users
+        :return: UserList
+        """
+        response = self._get(f"api/{self._uri}")
+        logger.debug("List all users successful")
+        data = response.json()
+
+        userlist = ast.literal_eval(data) if isinstance(data, str) else data
```

Comment:
Is this really needed ? If yes, a unit test must be added for the case when data is a string

Reply:
I think you can revert it

Reply:
reverted